base/applications/mspaint/dib.cpp

```diff
@@ -333,3 +333,61 @@ HBITMAP SkewDIB(HDC hDC1, HBITMAP hbm, INT nDegree, BOOL bVertical)
     DeleteDC(hDC2);
     return hbmNew;
 }
+
+typedef struct tagBITMAPINFODX
+{
+    BITMAPINFOHEADER bmiHeader;
+    RGBQUAD          bmiColors[256];
```

Comment:
Since in your code below, you copy bmiColors, but without necessarily always 256 elements, it looks like if you could instead use the standard `BITMAPINFO` structure instead of this custom one.

Reply:
Max 256. It have enough entries.
Some can be used (partially). The remainder can be ignored.

Reply:
`DIB_RGB_COLORS` sees `BITMAPINFO.bmiColors`.
The maximum number of colors on 8bpp is 256.
The `bmiColors` of standard `BITMAPINFO` is not enough.

Reply:
Fixed by using C++ inheritance:

```c
struct BITMAPINFODX : BITMAPINFO
{
    RGBQUAD bmiColorsAdditional[256 - 1];
};
```

Reply:
The "1" in bmiColors is an "ANYSIZE_ARRAY" to tell that data starts at bmiColors[0] and continue forwards.